src-gui/src/test/java/org/openkilda/service/StatsServiceTest.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.openkilda.config.ApplicationProperties;
+import org.openkilda.constants.Direction;
+import org.openkilda.exception.InvalidRequestException;
+import org.openkilda.integration.service.StatsIntegrationService;
+import org.openkilda.integration.service.SwitchIntegrationService;
+import org.openkilda.integration.source.store.SwitchStoreService;
+import org.openkilda.model.victoria.MetricValues;
+import org.openkilda.model.victoria.RangeQueryParams;
+import org.openkilda.model.victoria.Status;
+import org.openkilda.model.victoria.VictoriaData;
+import org.openkilda.model.victoria.dbdto.VictoriaDbData;
+import org.openkilda.model.victoria.dbdto.VictoriaDbRes;
+import org.openkilda.store.service.StoreService;
+import org.openkilda.test.MockitoExtension;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@ExtendWith(MockitoExtension.class)
+class StatsServiceTest {
```

Comment:
I would add more tests trying bad values to check the error handling.

Reply:
Thanks, added more corner cases for tests.